dune_client/api/extensions.py

```diff
@@ -0,0 +1,189 @@
+"""
+Extended functionality for the ExecutionAPI
+"""
+
+import time
+from typing import Union, Optional, Any
+
+from deprecated import deprecated
+
+from dune_client.api.execution import ExecutionAPI
+from dune_client.models import (
+    ResultsResponse,
+    DuneError,
+    ExecutionState,
+    QueryFailed,
+    ExecutionResultCSV,
+)
+from dune_client.query import QueryBase
+
+
+class ExtendedAPI(ExecutionAPI):
+    """
+    Extensions/helper methods built on top of the ExecutionAPI
```

Comment:
Provides higher level helper methods for faster and easier development on top of the base ExecutionAPI.

Reply:
```suggestion
    Provides higher level helper methods for faster and easier development on top of the base ExecutionAPI.
```